packages/test-utils/src/testUtils/appWrappers.test.tsx

```diff
@@ -14,22 +14,57 @@
  * limitations under the License.
  */
 
-import React from 'react';
+import React, { FC } from 'react';
 import { render } from '@testing-library/react';
-import { wrapInTestApp } from './appWrappers';
+import { wrapInTestApp, renderInTestApp } from './appWrappers';
 import { Route } from 'react-router';
+import { withLogCollector } from '@backstage/test-utils-core';
 
 describe('wrapInTestApp', () => {
-  it('should provide routing', () => {
-    const rendered = render(
-      wrapInTestApp(
-        <>
-          <Route path="/route1">Route 1</Route>
-          <Route path="/route2">Route 2</Route>
-        </>,
-        { routeEntries: ['/route2'] },
+  it('should provide routing and warn about missing act()', async () => {
+    const { error } = await withLogCollector(['error'], async () => {
+      const rendered = render(
+        wrapInTestApp(
+          <>
+            <Route path="/route1">Route 1</Route>
+            <Route path="/route2">Route 2</Route>
+          </>,
+          { routeEntries: ['/route2'] },
+        ),
+      );
+      expect(rendered.getByText('Route 2')).toBeInTheDocument();
+      await Promise.resolve();
```

Comment:
Why is this needed?

Reply:
It's waiting for the warnings to be generated, added a comment